result.go

```diff
@@ -0,0 +1,394 @@
+
+	for item, ok := next(); ok; item, ok = next() {
+		if compare(item, r) < 0 {
+			r = item
+		}
+	}
+
+	return
+}
+
+// Results iterates over a collection and returnes slice of interfaces
+func (q Query) Results() (r []interface{}) {
+	next := q.Iterate()
+
+	for item, ok := next(); ok; item, ok = next() {
+		r = append(r, item)
+	}
+
+	return
+}
+
+// SequenceEqual determines whether two collections are equal.
+func (q Query) SequenceEqual(q2 Query) bool {
+	next := q.Iterate()
+	next2 := q2.Iterate()
+
+	for item, ok := next(); ok; item, ok = next() {
+		item2, ok2 := next2()
+		if !ok2 || item != item2 {
+			return false
+		}
+	}
+
+	_, ok2 := next2()
+	return !ok2
+}
+
+// Single returns the only element of a collection, and nil
+// if there is not exactly one element in the collection.
+func (q Query) Single() interface{} {
+	next := q.Iterate()
+	item, ok := next()
+	if !ok {
+		return nil
+	}
+
+	_, ok = next()
+	if ok {
+		return nil
+	}
+
+	return item
+}
+
+// SingleWith returns the only element of a collection that satisfies
+// a specified condition, and nil if more than one such element exists.
+func (q Query) SingleWith(predicate func(interface{}) bool) (r interface{}) {
+	next := q.Iterate()
+	found := false
+
+	for item, ok := next(); ok; item, ok = next() {
+		if predicate(item) {
+			if found {
+				return nil
+			}
+
+			found = true
+			r = item
+		}
+	}
+
+	return
+}
+
+// SumInts computes the sum of a collection of numeric values.
+//
+// Values can be of any integer type: int, int8, int16, int32, int64.
+// The result is int64. Method returns zero if collection contains no elements.
+func (q Query) SumInts() (r int64) {
+	next := q.Iterate()
+	item, ok := next()
+	if !ok {
+		return 0
+	}
+
+	conv := getIntConverter(item)
+	r = conv(item)
+
+	for item, ok = next(); ok; item, ok = next() {
+		r += conv(item)
+	}
+
+	return
+}
+
+// SumUInts computes the sum of a collection of numeric values.
+//
+// Values can be of any unsigned integer type: uint, uint8, uint16, uint32, uint64.
+// The result is uint64. Method returns zero if collection contains no elements.
+func (q Query) SumUInts() (r uint64) {
+	next := q.Iterate()
+	item, ok := next()
+	if !ok {
+		return 0
+	}
+
+	conv := getUIntConverter(item)
+	r = conv(item)
+
+	for item, ok = next(); ok; item, ok = next() {
+		r += conv(item)
+	}
+
+	return
+}
+
+// SumFloats computes the sum of a collection of numeric values.
+//
+// Values can be of any float type: float32 or float64. The result is float64.
+// Method returns zero if collection contains no elements.
+func (q Query) SumFloats() (r float64) {
+	next := q.Iterate()
+	item, ok := next()
+	if !ok {
+		return 0
+	}
+
+	conv := getFloatConverter(item)
+	r = conv(item)
+
+	for item, ok = next(); ok; item, ok = next() {
+		r += conv(item)
+	}
+
+	return
+}
+
+// ToChannel iterates over a collection and outputs each element
+// to a channel, then closes it.
+func (q Query) ToChannel(result chan<- interface{}) {
```

Comment:
ah sorry I reviewed this 1 AM yesterday, didn't realize this is actually an input parameter. I thought it's a named return.